posthog/temporal/batch_exports/batch_exports.py

```diff
@@ -508,10 +532,44 @@ async def pause_batch_export_if_over_failure_threshold(
         settings.TEMPORAL_CLIENT_KEY,
     )
 
-    await sync_to_async(pause_batch_export)(
+    was_paused = await sync_to_async(pause_batch_export)(
         client, batch_export_id=batch_export_id, note="Paused due to exceeding failure threshold"
     )
-    return True
+
+    return was_paused
+
+
+async def cancel_running_backfills(batch_export_id: str) -> int:
+    """Cancel any running batch export backfills.
+
+    This is intended to be called once a batch export failure threshold has been exceeded.
+
+    Arguments:
+        batch_export_id: The ID of the batch export whose backfills will be cancelled.
+
+    Returns:
+        The number of cancelled backfills, if any.
+    """
+    client = await connect(
+        settings.TEMPORAL_HOST,
+        settings.TEMPORAL_PORT,
+        settings.TEMPORAL_NAMESPACE,
+        settings.TEMPORAL_CLIENT_ROOT_CA,
+        settings.TEMPORAL_CLIENT_CERT,
+        settings.TEMPORAL_CLIENT_KEY,
+    )
+
+    total_cancelled = 0
+
+    backfill_iter = iter_running_backfills_for_batch_export(uuid.UUID(batch_export_id))
+    backfills = await sync_to_async(list)(backfill_iter)  # type: ignore
```

Comment:
I hate this pattern. Getting rid of it as soon as one of the following happens:
* We update to django that supports async.
* We get rid of the ORM.

Reply:
on a side note, I think the django 4.2 update is complete, we should have `async for` for query iteration now

Reply:
Oh yeah, I see it from 2 days ago. I can take it for a spin. Thanks!